File: 36-valid-sudoku/valid-sudoku.java
class Solution {
    public boolean isValidSudoku(char[][] board) {
        ArrayList<String> seen = new ArrayList<>();

        for (int i = 0; i < 9; i++) {
            for (int j = 0; j < 9; j++) {
                char num = board[i][j];
                if (num == '.') continue; // Skip empty cells
                
                // Format for row, column, and grid checks
                String rowCheck = num + " in row " + i;
                String colCheck = num + " in col " + j;
                String boxCheck = num + " in box " + (i / 3) + "-" + (j / 3);

                // Check if the number already exists in the list
                if (seen.contains(rowCheck) || seen.contains(colCheck) || seen.contains(boxCheck)) {
                    return false; // Duplicate found
                }

                // Add checks to the list
                seen.add(rowCheck);
                seen.add(colCheck);
                seen.add(boxCheck);
            }
        }
        return true; // Valid Sudoku
    }
}

File: 36-valid-sudoku/valid-sudoku.py
class Solution(object):
    def isValidSudoku(self, board):
        """
        :type board: List[List[str]]
        :rtype: bool
        """
        seen = []  # Using a list instead of a set

        for i in range(9):
            for j in range(9):
                num = board[i][j]
                if num == '.':
                    continue  # Skip empty cells
                
                # Create strings using .format() instead of f-strings
                rowCheck = "{} in row {}".format(num, i)
                colCheck = "{} in col {}".format(num, j)
                boxCheck = "{} in box {}-{}".format(num, i // 3, j // 3)

                # Check if the number already exists in the list
                if rowCheck in seen or colCheck in seen or boxCheck in seen:
                    return False  # Duplicate found
                
                # Add constraints to the list
                seen.append(rowCheck)
                seen.append(colCheck)
                seen.append(boxCheck)

        return True  # Sudoku is valid
